File: cli/src/commands/new.rs
use anyhow::{anyhow, Result};
use crate::db::Database;
use crate::editor;
use crate::schema::SchemaValidator;

/// Execute the 'oak new' command
pub fn execute(db: &Database, validator: &SchemaValidator, name: &str) -> Result<()> {
    // Check if entry already exists
    if db.get_oak_entry(name)?.is_some() {
        return Err(anyhow!("Entry '{}' already exists. Use 'oak edit' to modify it.", name));
    }

    println!("Creating new Oak entry: {}", name);

    // Open editor with template
    let entry = editor::new_oak_entry(name, validator)?;

    // Save to database
    db.save_oak_entry(&entry)?;

    println!("✓ Successfully created entry for '{}'", entry.scientific_name);

    Ok(())
}

File: cli/cmd/new.go
package cmd

import (
	"fmt"

	"github.com/spf13/cobra"

	"github.com/jeff/oaks/cli/internal/editor"
)

var newCmd = &cobra.Command{
	Use:   "new <name>",
	Short: "Create a new Oak entry",
	Long:  `Creates a new Oak entry by opening your $EDITOR with a template.`,
	Args:  cobra.ExactArgs(1),
	RunE: func(cmd *cobra.Command, args []string) error {
		name := args[0]

		database, err := getDB()
		if err != nil {
			return err
		}
		defer database.Close()

		validator, err := getSchema()
		if err != nil {
			return err
		}

		// Check if entry already exists
		existing, err := database.GetOakEntry(name)
		if err != nil {
			return err
		}
		if existing != nil {
			return fmt.Errorf("oak entry '%s' already exists. Use 'oak edit' to modify it", name)
		}

		entry, err := editor.NewOakEntry(name, validator)
		if err != nil {
			return err
		}

		if err := database.SaveOakEntry(entry); err != nil {
			return err
		}

		fmt.Printf("Created oak entry: %s\n", entry.ScientificName)
		return nil
	},
}

func init() {
	rootCmd.AddCommand(newCmd)
}
